apps/ledger-live-desktop/src/newArch/Collectibles/components/DetailDrawer/components/ExternalViewerButton.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useCallback, memo, ReactElement } from "react";
+import styled from "styled-components";
+import { useHistory } from "react-router-dom";
+import { ExternalViewerButtonProps } from "LLD/Collectibles/types/DetailDrawer";
+import Box from "~/renderer/components/Box";
+import Button from "~/renderer/components/Button";
+import DropDownSelector, { DropDownItem } from "~/renderer/components/DropDownSelector";
+import IconExternal from "~/renderer/icons/ExternalLink";
+import useNftLinks from "LLD/Collectibles/hooks/useNftLinks";
+import { setDrawer } from "~/renderer/drawers/Provider";
+import { Icons } from "@ledgerhq/react-ui";
+
+const Separator = styled.div`
+  background-color: ${p => p.theme.colors.palette.divider};
+  height: 1px;
+  margin-top: 8px;
+  margin-bottom: 8px;
+`;
+
+const Item = memo(styled(DropDownItem)`
+  width: 100%;
+  cursor: pointer;
+  white-space: pre-wrap;
+  justify-content: space-between;
+  align-items: center;
+  display: flex;
+`);
+
+type Inner<A> = A extends Array<infer T> ? T : never;
+type Item = Inner<ReturnType<typeof useNftLinks>>;
+
+const ExternalViewerButtonComponent: React.FC<ExternalViewerButtonProps> = ({
+  nft,
+  account,
+  metadata,
+}: ExternalViewerButtonProps) => {
+  const history = useHistory();
+
+  const onHideCollection = useCallback(() => {
+    setDrawer();
+    history.replace(`/account/${account.id}/`);
+  }, [account.id, history]);
+
+  const items = useNftLinks(account, nft, metadata, onHideCollection);
+
+  const renderItem = ({ item }: { item: Item }): ReactElement => {
+    if (item.type === "separator") {
+      return <Separator />;
+    }
+
+    const Icon = item.Icon
+      ? // TODO: the icons have incompatible props (size: string / number)
+        // eslint-disable-next-line
+        React.createElement(item.Icon as any, {
+          size: 16,
+        })
+      : null;
```

Comment:
You can maybe update with new Icons and then update types?

Reply:
https://ledgerhq.atlassian.net/browse/LIVE-12879